### PR TITLE
[improve][ci] skip CI cronjob in forks 

### PR DESCRIPTION
**Pull request by tisonkun**
### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/tisonkun/pulsar/pull/14

### CODE REVIEW

**Review by nicoloboschi — APPROVED** [review]
good catch! 
I didn't realized the schedule would run in every single fork of Pulsar every day, that is such a waste of GH resources.

**Comment by tisonkun** [comment]
@nicoloboschi Thanks for your review! I push one more commit e861b6bda23bd010fd5cd34ee381c6ac8eff9f75 to skip the always running `pulsar-ci-checks-completed` job. I'm unsure whether it's the proper way, perhaps we can alternatively check the `needs.*.result` but I found the pushed way is shorter.

**Comment by tisonkun** [comment]
It's ready to merge.

cc @lhotari @nicoloboschi please take a look.

File: .github/workflows/pulsar-ci.yaml [hunk 1]
```diff
@@ -953,11 +954,11 @@ jobs:
   # It cleans up the binaries in the same job in order to not spin up another runner for basically doing nothing.
   pulsar-ci-checks-completed:
     name: "Pulsar CI checks completed"
-    if: always()
+    if: ${{ !cancelled() }}
```
Comment (nicoloboschi) [inline .github/workflows/pulsar-ci.yaml:957]: I'm not sure. If I cancel the workflow manually then this job is marked as skipped and then GH allows me merge the pull. Could you verify it ?

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by tisonkun** [comment]
@nicoloboschi You're right. I recover the `always() && ...` pattern and it seems the issue you reported vanished.

If we can reach a consensus on this version, let's see how the next cronjob works :)

**Comment by tisonkun** [comment]
ping @nicoloboschi @lhotari

**Review by lhotari — APPROVED** [review]

**Review by nicoloboschi — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your reviews! Merging...

**Pull request merged**